lib/redirect.js

```diff
@@ -145,7 +147,7 @@ Redirect.prototype.onResponse = function (response) {
 
   request.emit('redirect')
 
-  request.init()
+  request.init(request.options)
```

Comment:
I think that re-calling `init` with original `options` is better. Well, is this bad idea?